File: DynamicKey/AgoraDynamicKey/deno/sample/FpaTokenBuilderSample.js
import { FpaTokenBuilder } from "../src/FpaTokenBuilder.js";

// Need to set environment variable AGORA_APP_ID
const appId = Deno.env.get("AGORA_APP_ID");
// Need to set environment variable AGORA_APP_CERTIFICATE
const appCertificate = Deno.env.get("AGORA_APP_CERTIFICATE");

console.log("App Id:", appId);
console.log("App Certificate:", appCertificate);
if (appId == undefined || appId == "" || appCertificate == undefined || appCertificate == "") {
    console.log("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE");
    Deno.exit(1);
}

let token = FpaTokenBuilder.buildToken(appId, appCertificate);
console.log("Token with FPA service:", token);

File: DynamicKey/AgoraDynamicKey/dart/sample/fpa_token_builder_sample.dart
import 'dart:io';

import 'package:agora_token_generator/agora_token_generator.dart';

/// Generates an FPA login Token007 token.
void main() {
  final appId = Platform.environment['AGORA_APP_ID'] ?? '';
  final appCertificate = Platform.environment['AGORA_APP_CERTIFICATE'] ?? '';

  if (appId.isEmpty || appCertificate.isEmpty) {
    stderr.writeln(
      'Need to set environment variable AGORA_APP_ID and '
      'AGORA_APP_CERTIFICATE',
    );
    exitCode = 1;
    return;
  }

  final token = FpaTokenBuilder.buildToken(
    appId: appId,
    appCertificate: appCertificate,
  );
  if (token.isEmpty) {
    throw StateError('FPA token generation failed');
  }
  stdout.writeln('FPA token: $token');
}
